tools/snapshot/unittests/export_test.go

```diff
@@ -0,0 +1,96 @@
+package unittests
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"xrplf/clio/clio_snapshot/internal/export"
+	"xrplf/clio/clio_snapshot/mocks"
+	pb "xrplf/clio/clio_snapshot/org/xrpl/rpc/v1"
+
+	"github.com/golang/mock/gomock"
+)
+
+func TestExportDeltaLedgerData(t *testing.T) {
+
+	tests := []struct {
+		name     string
+		startSeq uint32
+		endSeq   uint32
+	}{
+		{"OneSeq", 1, 1},
+		{"MultipleSeq", 1, 20},
+		{"EndSeqLessThanStartSeq", 20, 1},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockClient := mocks.NewMockXRPLedgerAPIServiceClient(ctrl)
+
+			mockResponse := &pb.GetLedgerResponse{}
+
+			times := tt.endSeq - tt.startSeq + 1
+			if tt.endSeq < tt.startSeq {
+				times = 0
+			}
+
+			mockClient.EXPECT().GetLedger(gomock.Any(), gomock.Any()).Return(mockResponse, nil).Times(int(times))
+
+			defer os.RemoveAll("test")
+
+			export.ExportFromDeltaLedgerImpl(mockClient, tt.startSeq, tt.endSeq, "test")
+
+			_, err := os.Stat("test")
+
+			assert.Equal(t, os.IsNotExist(err), tt.endSeq < tt.startSeq)
+		})
+	}
+}
+
+func TestExportFullLedgerData(t *testing.T) {
```

Comment:
These tests look almost like integration tests. Shouldn't there be unit tests for each function where possible?

Reply:
I only test the public interfaces. I keep the tests under unittests module, not the export. I don't want to change functions to private just for testing purpose.